### PR TITLE
feat(login) : implement login with set cookie

### PR DESCRIPTION
**Pull request by kibo-susantasamanta**
@kibo-kevinwatts @kibo-sushant @kibo-chandradeeptalaha

### CODE REVIEW

File: composables/useUser/index.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  const login = async (params: LogInUserParams) => {
+    const user = {
+      loginInput: {
+        username: params.user.username,
+        password: params.user.password,
+      },
+    }
+    try {
+      loading.value = true
+      const customerAccount = await fetcher({
```
Comment (kibo-kevinwatts) [inline composables/useUser/index.ts]: change customerAccount variable to "response"

File: composables/useUser/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        query: loginMutation,
+        variables: user,
+      })
+      if (customerAccount.data.account && customerAccount.data.account.userId) {
```
Comment (kibo-kevinwatts) [inline composables/useUser/index.ts]: simplify this
if(customerAccount?.data?.account?.userId)

File: composables/useUser/index.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        // set cookie
+        const cookie = {
+          accessToken: account.accessToken,
+          accessTokenExpiration: account.accessTokenExpiration,
+          refreshToken: account.refreshToken,
+          refreshTokenExpiration: account.refreshToken,
+          userId: account?.refreshToken,
+        }
+        storeClientCookie(authCookieName, cookie)
+        load()
```
Comment (kibo-kevinwatts) [inline composables/useUser/index.ts]: await load()

File: components/AppHeader.vue [hunk 4]
```diff
@@ -69,10 +68,19 @@ export default {
   directives: { clickOutside },
   // setup(props, { root }) {
   setup() {
+    const { user, load: loadUser } = useUser()
     const isSearchOpen = ref(false)
+    const isAuthenticated = computed(() => {
+      return userGetters.isLoggedInUser(user.value)
+    })
     const result = ref(null)
+    const router = ref()
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: router ref not needed

File: components/AppHeader.vue [hunk 5]
```diff
@@ -69,10 +68,19 @@ export default {
...
+    const instance = getCurrentInstance()
+    router.value = instance?.root.type.router
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: no need to assign router.value here

File: components/AppHeader.vue [hunk 6]
```diff
@@ -96,23 +104,44 @@ export default {
...
     return {
+      user,
+      loadUser,
       accountIcon,
       isSearchOpen,
       handleSearch,
       isMobile,
       removeSearchResults,
       handleAccountClick,
+      router,
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: remove router in return here

File: components/AppHeader.vue [hunk 7]
```diff
@@ -96,23 +104,44 @@ export default {
...
+        return router.value.push("/my-account")
+      }
       toggleLoginModal()
     }
 
+    useAsync(async () => {
+      await loadUser()
+    }, null)
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (kibo-kevinwatts) [inline components/AppHeader.vue]: remove commented out code

File: server/helpers/nodeCookieHelpers.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
```
Comment (kibo-kevinwatts) [inline server/helpers/nodeCookieHelpers.ts]: We don't need this file

File: server/helpers/sessionTokenHelpers.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { UserAuthTicket } from "@kibocommerce/graphql-client"
+import type { KiboIncomingMessage, Response } from "../types/Api"
+import { COOKIES } from "../../constants"
```
Comment (kibo-kevinwatts) [inline server/helpers/sessionTokenHelpers.ts]: we don't need this module

File: constants.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line import/prefer-default-export
+export const COOKIES = {
```
Comment (kibo-kevinwatts) [inline constants.ts:2]: we don't need this

File: gql/queries/getCurrentUser.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
+export const getCurrentUser = `
+  query getUser {
+    customerAccount: getCurrentAccount {
+      id
+      firstName
+      lastName
+      emailAddress
+      userName
```
Comment (kibo-kevinwatts) [inline gql/queries/getCurrentUser.ts:9]: Include userId

File: gql/mutations/user/loginMutation.ts [hunk 12]
```diff
@@ -0,0 +1,42 @@
+export const loginMutation = `
+mutation login($loginInput:CustomerUserAuthInfoInput!) {
+  account:createCustomerAuthTicket(customerUserAuthInfoInput:$loginInput) {
+    customerAccount {
+      emailAddress
```
Comment (kibo-kevinwatts) [inline gql/mutations/user/loginMutation.ts]: Return same properties for customerAccount as the "getCurrentUser" query

File: components/LoginModal.vue [hunk 13]
```diff
@@ -168,12 +161,22 @@ export default {
...
   },
   setup() {
+    const { user, login, loading, error: userError, isAuthenticated } = useUser()
     const { isLoginModalOpen, toggleLoginModal } = useUiState()
     const form = ref({})
     const isLogin = ref(false)
     const isForgotten = ref(false)
     const rememberMe = ref(false)
     const createAccount = ref(false)
+    const error = reactive({
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: Why create a new reactive reference for error?
Can we just use userError?

File: components/LoginModal.vue [hunk 14]
```diff
@@ -200,23 +203,48 @@ export default {
       isForgotten.value = value
     }
 
+    const handleForm = (fn) => async () => {
+      resetErrorValues()
+      await fn({ user: form.value })
+      const hasUserErrors = userGetters.hasLoginError(userError.value)
+      if (hasUserErrors) {
+        const loginError = userError.value
+        error.login = loginError.login?.extensions?.response?.body?.message
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: set errors (if any) inside composable after api call

File: components/LoginModal.vue [hunk 15]
```diff
@@ -200,23 +203,48 @@ export default {
...
+        error.register = loginError.register?.message
+        return user
```
Comment (kibo-kevinwatts) [inline components/LoginModal.vue]: no need to return user

File: composables/useUser/index.ts [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+      })
+      user.value = response.data?.customerAccount
+    } catch (err) {
+    } finally {
+      loading.value = false
+    }
+  }
+
+  // User
+  const login = async (params: LogInUserParams) => {
```
Comment (kibo-kevinwatts) [inline composables/useUser/index.ts]: LogInUserParams can just be CustomerUserAuthInfoInput

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: components/AppHeader.vue (hunk 7 above)
Comment (kibo-susantasamanta) [reply components/AppHeader.vue]: Removed.

File: components/LoginModal.vue (hunk 15 above)
Comment (kibo-susantasamanta) [reply components/LoginModal.vue]: Fixed

File: composables/useUser/index.ts (hunk 1 above)
Comment (kibo-susantasamanta) [reply composables/useUser/index.ts]: resolved

File: composables/useUser/index.ts (hunk 3 above)
Comment (kibo-susantasamanta) [reply composables/useUser/index.ts]: resolved

File: composables/useUser/index.ts (hunk 2 above)
Comment (kibo-susantasamanta) [reply composables/useUser/index.ts]: simplified

File: components/AppHeader.vue (hunk 5 above)
Comment (kibo-susantasamanta) [reply components/AppHeader.vue]: resolved

File: components/AppHeader.vue (hunk 4 above)
Comment (kibo-susantasamanta) [reply components/AppHeader.vue]: used useNuxtApp instead...

File: components/LoginModal.vue (hunk 15 above)
Comment (kibo-susantasamanta) [reply components/LoginModal.vue]: fixed

File: gql/mutations/user/loginMutation.ts (hunk 12 above)
Comment (kibo-susantasamanta) [reply gql/mutations/user/loginMutation.ts]: fixed.

File: components/AppHeader.vue (hunk 6 above)
Comment (kibo-susantasamanta) [reply components/AppHeader.vue]: Removed. Used useNuxtApp instead.

File: server/helpers/nodeCookieHelpers.ts (hunk 8 above)
Comment (kibo-susantasamanta) [reply server/helpers/nodeCookieHelpers.ts]: Removed.

File: constants.ts (hunk 10 above)
Comment (kibo-susantasamanta) [reply constants.ts:2]: Removed.

File: gql/queries/getCurrentUser.ts (hunk 11 above)
Comment (kibo-susantasamanta) [reply gql/queries/getCurrentUser.ts:9]: UserId Included.

File: server/helpers/sessionTokenHelpers.ts (hunk 9 above)
Comment (kibo-susantasamanta) [reply server/helpers/sessionTokenHelpers.ts]: Module removed.

File: composables/useUser/index.ts (hunk 16 above)
Comment (kibo-susantasamanta) [reply composables/useUser/index.ts]: used CustomerUserAuthInfoInput instead LogInUserParams

File: components/LoginModal.vue (hunk 13 above)
Comment (kibo-susantasamanta) [reply components/LoginModal.vue]: we can use useError as well.

File: components/LoginModal.vue (hunk 14 above)
Comment (kibo-susantasamanta) [reply components/LoginModal.vue]: resolved.

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**